pkg/cloud/services/compute/bastion.go

```diff
@@ -36,6 +39,21 @@ func (s *Service) DeleteBastion(openStackCluster *infrav1.OpenStackCluster, serv
 		return err
 	}
 
+	err = util.PollImmediate(RetryIntervalInstanceStatus, TimeoutInstanceDelete, func() (bool, error) {
+		_, err = s.GetInstance(instance.ID)
+		if err != nil {
+			if capoerrors.IsNotFound(err) {
+				return true, nil
+			}
+			return false, err
+		}
+		return false, nil
+	})
+	if err != nil {
+		record.Warnf(openStackCluster, "FailedDeleteServer", "Failed to deleted server %s: %v", instance.ID, err)
```

Comment:
looks like we also add the name in l.38, maybe good to also add it here for consistency